packages/format-library/src/code/index.js

```diff
@@ -20,16 +20,18 @@ export const code = {
 	__unstableInputRule( value ) {
 		const BACKTICK = '`';
 		const { start, text } = value;
-		const characterBefore = text.slice( start - 1, start );
+		const characterBefore = text[ start - 1 ];
```

Comment:
Maybe we should add `characterBefore` as one of the parameters of `__unstableInputRule` because almost all rules will need this.

Maybe even better yet, associate the callback with a trigger character so we can built this in? There's a reason why this API is not stable 😅

Reply:
maybe, but do we know this is the only use-case? apart from inline code I don't know what the purpose is here. would we ever want to run this on triple-backtick code fences?

I guess at some point _most_ replacements would have a stable trigger character, unless we wanted to re-use code for something like mentions.

in any case `text[ start - 1 ]` is essentially free so I don't see it causing harm to leave in before we have a good feeling for what this API should end up looking like.